platform-validation/scripts/compatibility.sh

```diff
@@ -0,0 +1,418 @@
+                ;;
+            aarch64)
+                _cputype=armv7
+                if [ "$_ostype" = "linux-android" ]; then
+                    _ostype=linux-androideabi
+                else
+                    _ostype="${_ostype}eabihf"
+                fi
+                ;;
+        esac
+    fi
+
+    # Detect armv7 but without the CPU features Rust needs in that build,
+    # and fall back to arm.
+    # See https://github.com/rust-lang/rustup.rs/issues/587.
+    if [ "$_ostype" = "unknown-linux-gnueabihf" ] && [ "$_cputype" = armv7 ]; then
+        if ensure grep '^Features' /proc/cpuinfo | grep -q -v neon; then
+            # At least one processor does not have NEON.
+            _cputype=arm
+        fi
+    fi
+
+    OSTYPE=$_ostype
+    ARCH=$_cputype
+}
+
+perform_cleanup(){
+    rm -rf cap.txt || true
+    #TODO : Cleanup docker images
+}
+
+
+# ------------------------------------------------------------------------------
+# Check whether the Target Device can be used to set capability. EdgeHub Runtime component sets CAP_NET_BIND which is Required
+# for Azure IoT Edge Operation
+# ------------------------------------------------------------------------------
+perform_capability_check_host(){
+    
+    wrap_debug "Checking Set Cap Capability on Host.."
+    touch cap.txt
+    setcap "cap_net_bind_service=+ep" cap.txt
+    if [ $? != 0 ]; then
+        #TODO Check Mark Failed in Red
+        wrap_debug "Failed to Set Capability on Host Container"
+        wrap_fail "Capability_Check_Host" "Fail"
+        return
+    fi
+
+    contains=$(getcap cap.txt | grep 'cap_net_bind_service+ep')
+    if [ $? != 0 ] && [ -z "${contains##*cap_net_bind_service+ep*}" ]; then
+        wrap_debug "Failed to Verify Set Capability on Host Container"
+        wrap_fail "Capability_Check_Host" "Fail"
+        return
+    fi
+
+    wrap_pass "Capability_Check_Host" "Pass"
+}
+
+perform_capability_check_container(){
+    #Check For Docker
+    wrap_debug "Checking Set Cap Capability on Container.."
+    need_cmd docker
+    CAP_CMD="getcap cap.txt"
+    DOCKER_VOLUME_MOUNTS=''
+    DOCKER_IMAGE="ubuntu:18.04"
+    #TODO: Use different images based on platform and arch
+    docker run --rm \
+    --user root \
+    -e 'USER=root' \
+    -i \
+    $DOCKER_VOLUME_MOUNTS \
+    "$DOCKER_IMAGE" \
+    sh -c "
+        export DEBIAN_FRONTEND=noninteractive
+        set -e &&
+        apt-get update 1>/dev/null 2>&1 &&
+        apt-get install -y libcap2-bin 1>/dev/null 2>&1
+        touch cap.txt
+        setcap 'cap_net_bind_service=+ep' cap.txt
+        if [ $? != 0 ]; then
+            exit 1
+        fi
+    "
+    if [ $? != 0 ]; then
+        #TODO Check Mark Failed in Red
+        wrap_debug "Failed to Check Capability in Container, Check Failed"
+        wrap_fail "Capability_Check_Container" "Fail"
+        return
+    fi
+    wrap_pass "Capability_Check_Container" "Pass"
+}
+
+
+echo "Running IoT Edge Compatibility Tool"
```

Comment:
I don't think we need this "Running..." line, since the user issues the command to run it.

Reply:
Removed